crates/uv/src/commands/project/mod.rs

```diff
@@ -101,7 +101,7 @@ pub(crate) enum ProjectError {
 pub(crate) fn find_requires_python(
     workspace: &Workspace,
 ) -> Result<Option<RequiresPython>, uv_resolver::RequiresPythonError> {
-    RequiresPython::union(workspace.packages().values().filter_map(|member| {
+    RequiresPython::intersection(workspace.packages().values().filter_map(|member| {
```

Comment:
We can warn here, if people feel strongly. I'm not sure that we should, though, since users will see it _every_ time without any way to resolve it.